runners/direct-java/src/main/java/org/apache/beam/runners/direct/StructuralKey.java

```diff
@@ -26,52 +26,83 @@
  * A (Key, Coder) pair that uses the structural value of the key (as provided by
  * {@link Coder#structuralValue(Object)}) to perform equality and hashing.
  */
-class StructuralKey<K> {
+abstract class StructuralKey<K> {
+
+  private StructuralKey() {
+    // Do not extend outside of this class
+  }
+
+  public abstract K getKey();
+  /**
+   * Get the empty {@link StructuralKey}. All instances of the empty key are considered equal.
```

Comment:
I was under the impression that all instances of the empty key should be considered distinct